File: aoc-2015/src/test/java/com/capital7software/aoc2015/days/Day22Test.java
package com.capital7software.aoc2015.days;

import com.capital7software.aoc.aoc2015.days.Day22;
import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day22Test extends AdventOfCodeTestBase {
    private static final Logger LOGGER = LoggerFactory.getLogger(Day22Test.class);

    @BeforeEach
    void setUp() {
        var instance = new Day22();
        setupFromFile(instance.getDefaultInputFilename());
    }

    @Test
    void leastAmountOfManaAndStillWin() {
        var instance = new Day22();
        int expected = 953;
        var actual = instance.leastAmountOfManaAndStillWin(lines, false);
        assertEquals(expected, actual, "The least amount of mana " +
                "is not what was expected: " + expected);
    }

    @Test
    void leastAmountOfManaOnHardAndStillWin() {
        var instance = new Day22();
        int expected = 1289;
        var actual = instance.leastAmountOfManaAndStillWin(lines, true);
        assertEquals(expected, actual, "The least amount of mana on hard " +
                "is not what was expected: " + expected);
    }

    @Override
    protected Logger getLogger() {
        return LOGGER;
    }
}

File: aoc-2017/src/test/kotlin/com/capital7software/aoc/aoc2017aoc/days/Day22Test.kt
package com.capital7software.aoc.aoc2017aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day22Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day22Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day22()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testRunBurstsSmall() {
    val instance = Day22()
    val bursts = 70
    val expected = Pair(41, 29)
    val actual = instance.runBursts(lines, bursts)

    Assertions.assertEquals(
        expected, actual, "$actual results of $bursts bursts " +
        "is not what was expected: $expected"
    )
  }

  @Test
  fun testRunBursts() {
    val instance = Day22()
    val bursts = 10_000
    val expected = Pair(5_587, 4_413)
    val actual = instance.runBursts(lines, bursts)

    Assertions.assertEquals(
        expected, actual, "$actual results of $bursts bursts " +
        "is not what was expected: $expected"
    )
  }

  @Test
  fun testRunBurstsNewLogicSmall() {
    val instance = Day22()
    val bursts = 100
    val expected = Pair(26, 74)
    val actual = instance.runBursts(lines, bursts, true)

    Assertions.assertEquals(
        expected, actual, "$actual results of $bursts bursts " +
        "is not what was expected: $expected"
    )
  }

  @Test
  fun testRunBurstsNewLogic() {
    val instance = Day22()
    val bursts = 10_000_000
    val expected = Pair(2_511_944, 7_488_056)
    val actual = instance.runBursts(lines, bursts, true)

    Assertions.assertEquals(
        expected, actual, "$actual results of $bursts bursts " +
        "is not what was expected: $expected"
    )
  }
}
